src/data/release.rs

```diff
@@ -23,24 +28,43 @@ impl Release {
         })
     }
 
-    async fn fetch_manifest(url: Option<&str>) -> Result<Option<DistManifest>> {
-        if let Some(manifest_url) = url {
-            match reqwest::get(manifest_url).await?.error_for_status() {
-                Ok(resp) => match resp.json::<DistManifest>().await {
-                    Ok(manifest) => Ok(Some(manifest)),
-                    Err(e) => {
-                        let msg = format!("Failed to parse dist-manifest at {manifest_url}.\nDetails:{e}\n\nSkipping...");
-                        Message::new(MessageType::Warning, &msg).print();
-                        Ok(None)
-                    }
-                },
-                Err(e) => Err(OrandaError::CargoDistManifestFetchError {
-                    url: manifest_url.to_string(),
-                    status_code: e.status().unwrap_or(reqwest::StatusCode::BAD_REQUEST),
-                }),
-            }
+    async fn fetch_manifest(
+        gh_release: &GithubRelease,
+        repo: &GithubRepo,
+    ) -> Result<Option<DistManifest>> {
+        let tag = &gh_release.tag_name;
+        if gh_release.has_dist_manifest() {
+            let request = octolotl::request::ReleaseAsset::new(
+                &repo.owner,
+                &repo.name,
+                tag,
+                cargo_dist::MANIFEST_FILENAME,
+            );
+            let response = octolotl::Request::send(&request, true)
+                .await?
+                .error_for_status()?;
+
+            Ok(Self::parse_response(response, tag).await?)
         } else {
             Ok(None)
         }
     }
+
+    async fn parse_response(
+        response: reqwest::Response,
+        tag: &str,
+    ) -> Result<Option<DistManifest>> {
+        let res: serde_json::Value = serde_json::from_str(&response.text().await?)?;
+        let pretty_response = serde_json::to_string_pretty(&res)?;
+        Ok(
+            match SourceFile::new("", pretty_response).deserialize_json::<DistManifest>() {
+                Ok(manifest) => Some(manifest),
+                Err(e) => {
+                    let msg = format!("Failed to parse dist-manifest for release {tag}.\nDetails:{e}\n\nSkipping...");
```

Comment:
Wait did this actually get miette printing? I thought you had to wrap the `e` in a Report and debug print that.

Reply:
Although to do that pleasantly you likely want to just make this whole message a miette error (well, warning) and print the whole thing with miette...

Reply:
so i did the impl for this but when i experienced it- it didn't feel right. we don't want the user to "fix" the error- they can't really- it would require going back and republishing old incompatible releases. i think the single line for "information" is the correct move here...

```
↪ >o_o< INFO: Starting dev, looking for paths to watch...
↪ >o_o< INFO: Loading config at ./oranda.json
↪ >o_o< INFO: Detected Rust project...
↪ >o_o< INFO: Found 5 paths to watch, starting watch...
↪ >o_o< DEBUG: Files watched: ["README.md", "./oranda.json", "./docs/book.toml", "./docs/src", "/Users/ag_dubs/axo/oranda/Cargo.toml"]
↪ >o_o< INFO: Running build...
↪ >o_o< INFO: Loading config at ./oranda.json
↪ >o_o< INFO: Detected Rust project...
↪ >o_o< INFO: Building nav...
↪ >o_o< INFO: Adding artifacts page...
↪ >o_o< INFO: Adding book...
↪ >o_o< INFO: Adding changelog...
  ⚠ Failed to parse dist-manifest for release v0.0.1-prerelease1. Skipping...
  ├─▶   × failed to parse JSON
  │      ╭─[:7:1]
  │    7 │       "artifacts": [
  │    8 │         {
  │      ·        ─
  │    9 │           "assets": [
  │      ╰────
  │
  ╰─▶ invalid type: map, expected a string at line 8 column 8

↪ >o_o< INFO: Detected pre-releases...
↪ >o_o< INFO: Building mdbook...
✓ >o_o< SUCCESS: Successfully built your site in the `public` directory. To view, run `oranda serve`.
↪ >o_o< INFO: Running serve...
↪ >o_o< INFO: Loading config at ./oranda.json
↪ >o_o< INFO: Detected Rust project...
↪ >o_o< INFO: Found build in public directory...
✓ >o_o< SUCCESS: Your project is available at: http://127.0.0.1:7979/oranda
```